### ISSUE TITLE
ability to configure option "Automatically publish a replay when your live ends" by default for all streamings

### ISSUE THREAD

**Issue by pedro-nonfree** [Chocobozzz/PeerTube#4024]
Using the administration configuration (Administration -> Configuration -> Live Streaming) add an option that enables by default the option "Automatically publish a replay when your live ends". Because right now, specific use cases need to ensure constantly that the option is marked

**Comment by tio-trom** [comment]
Yah I lost a 2h recording because I forgot to enable that....most users won't even know you have to enable such a thing. Normally if you do a live you also want to stored. You can always delete afterwards....I am so sad I lost our podcast. Tried so hard to invite this researcher and discuss about the online conspiracy theories and videos posted on Peertube and such, from an academic perspective. Now all lost. I accuse myself for it, but I also think Peertube played a role.

**Comment by futureisfoss** [comment]
No offense, but it seems like a really dumb move to make this setting turned off by default. If all the user wanted was a simple live stream, there are other software for that. Peertube is a video sharing platform, most people will expect the live stream to be saved afterwards.

I request you to prioritize this issue, it shouldn't be hard to fix but the benefits are huge.

**Comment by ROBERT-MCDOWELL** [comment]
@futureisofss
PR welcome!

**Comment by Chocobozzz** [comment]
> No offense, but it seems like a really dumb move to make this setting turned off by default. If all the user wanted was a simple live stream, there are other software for that. Peertube is a video sharing platform, most people will expect the live stream to be saved afterwards.

We don't enable save replay by default because it could terminate the stream if user quota is exceeded. Then someone would probably have said that it was a dumb move to automatically enable an option users may miss and than can kill the live stream.

So I don't have the perfect solution. Some workaround I'm thinking about:
 * Display a message near the publish button to notify users the stream won't be saved
 * Force users to specify if they want or not save the stream (so disabling publish until they fill the input)

**Comment by tio-trom** [comment]
I think any of those two options are better indeed. I understand a bit more your reasoning now.

**Comment by tio-trom** [comment]
Make two options to force users when they create a live, with simple wording:

1. Livestream and Save (stream and save the stream on your account)
2. Livestream and Delete (stream and delete the stream after it ends)

The current wording is somewhat confusing in my opinion. Check it out:

1. This is a permanent live (You can stream multiple times in a permanent live. The URL for your viewers won't change but you cannot save replays of your lives)
2. Automatically publish a replay when your live ends (⚠️ If you enable this option, your live will be terminated if you exceed your video quota)

I am not sure what is the use of the permanent live....are people using it? An URL where you can stream multiple times at, but never save the replay. Considering Peertube does not have a livechat or the ability to notify users when you go live, it this option useful? I am simply asking not at all criticizing.

**Comment by pedro-nonfree** [comment]
all options are good to improve UX and avoid confusing among users. Consider please also the default option for cases where quota is infinite, there are few users, and they just want the stream stored always without to worry if they have checked or unchecked that option

permanent live is important if you cannot guarantee stable internet connection and the streaming share link is important. A little-small connection drop will end the live and you have to create a new one and inform your users. So saving replays on permanent live in #3396 is also very relevant for a solid streaming service.

... Also (again) the ability to put default options so users only have to do the live itself without worrying about the details of it, because that particular service.com works always that way

**Comment by futureisfoss** [comment]
> I think any of those two options are better indeed. I understand a bit more your reasoning now.
> 
> EDIT: Although wouldn't it be better to warn users that if their stream exceeds their quota, the stream will be terminated? Like leave it default to be stored and add a warning and the ability to op-out of storing the stream. Something like: "You have XGB left on your account, if the livestream exceeds your quota, the stream will be terminated." And the ability to opt-out where that warning appears. Seems more reasonable to me.
> 
> But overall I think forcing users to take a decision is for sure the way to go.

I agree with this approach. I think its better to force users to take a decision about this, and show them proper warnings where needed. Another thing I want to add is that, it'll be nice if a user can set a default option on his account whether or not to save his livestreams. I think giving this choice to individual users on their account settings is better than giving it to the instance admin only. Also in this case, show them proper warnings as to how much quota they have left etc.

**Comment by Chocobozzz** [comment]
I forced the live type selection in the first step "go live" form in https://github.com/Chocobozzz/PeerTube/commit/1e2fe802d1f82b81ac30fd9ca267da12f75032a2

![screen_2021-10-15-11:38:16](https://user-images.githubusercontent.com/5180488/137468058-e329c824-e9f6-4269-9987-9b177442d220.png)
